src/coreclr/vm/assembly.cpp

```diff
@@ -686,6 +686,21 @@ Assembly *Assembly::CreateDynamic(AppDomain *pDomain, ICLRPrivBinder* pBinderCon
             pAssem->SetIsTenured();
             pRetVal = pAssem;
         }
+
+        {
+            // We should be to create dependency link between LoaderAllocator of the dynamic assembly
+            // and the LoaderAllocator of the ALC if they are both unloadable.
+            if (pBinderContext != nullptr && createdNewAssemblyLoaderAllocator && pLoaderAllocator->IsCollectible())
+            {
+                AssemblyLoaderAllocator* pAssemblyLoaderAllocator = nullptr;
+                pBinderContext->GetLoaderAllocator((LPVOID*)&pAssemblyLoaderAllocator);
+
+                if (pAssemblyLoaderAllocator->IsCollectible())
+                {
+                    _ASSERTE(pLoaderAllocator->EnsureReference(pAssemblyLoaderAllocator) == TRUE);
```

Comment:
This would compile to nothing in release build (the whole _ASSERTE is ignored).

Reply:
Removed _ASSERTE